packages/paima-sdk/paima-mw-core/src/helpers/posting.ts

```diff
@@ -213,7 +213,10 @@ export async function postConciselyEncodedData(
 }
 
 async function getAdjustedHeight(deploymentChainBlockHeight: number): Promise<number> {
-  const emulatedActive = getEmulatedBlocksActive() ?? (await emulatedBlocksActiveOnBackend());
+  const activeOnBackend = await emulatedBlocksActiveOnBackend();
+  const emulatedActive =
+    getEmulatedBlocksActive() ?? (activeOnBackend.success && activeOnBackend.result);
```

Comment:
this is a performance regression I think. The code on the left only calls the endpoint is `getEmulatedBlocksActive` is true, but the code on the right always calls the endpoint

Reply:
yeah, good point, I didn't consider the order. I'll fix that.